### ISSUE TITLE
Make the Checkbox validationHint consistent with other form components

### ISSUE THREAD

**Issue by robinmetral** [sumup-oss/circuit-ui#1768]
## Component to amend

`Checkbox`

## Visual

We're missing Figma specs for this component. Suggested changes will be reviewed by design

## Context

Currently, a Checkbox `validationHint` (description or warning/error message) is rendered as a tooltip. Beyond the fact that the current `Tooltip` component isn't accessible (see #1676), this is also visually inconsistent and has inherent accessibility issues (descriptions and status messages should not be concealed in a tooltip, they contain important information that helps users fill forms)

**Comment by connor-baer** [comment]
Closed by #1789
